cpp/benchmarks/common/generate_input.cu

```diff
@@ -642,6 +676,42 @@ std::unique_ptr<cudf::table> create_random_table(std::vector<cudf::type_id> cons
   return std::make_unique<cudf::table>(std::move(output_columns));
 }
 
+std::unique_ptr<cudf::table> create_sequence_table(std::vector<cudf::type_id> const& dtype_ids,
+                                                   cudf::size_type num_cols,
+                                                   row_count num_rows,
+                                                   float null_probability,
+                                                   unsigned seed)
+{
+  auto const out_dtype_ids = repeat_dtypes(dtype_ids, num_cols);
+  auto columns             = std::vector<std::unique_ptr<cudf::column>>(num_cols);
+  auto init                = cudf::make_default_constructed_scalar(cudf::data_type{dtype_ids[0]});
+  if (dtype_ids.size() == 1) {
+    std::generate_n(columns.begin(), num_cols, [&]() mutable {
+      auto col = cudf::sequence(num_rows.count, *init);
+      if (null_probability >= 0.0) {
+        auto [mask, count] = create_random_null_mask(num_rows.count, null_probability, seed++);
+        col->set_null_mask(std::move(mask), count);
+      }
+      return col;
```

Comment:
The difference between 2 lambda, is that first lambda has `init` common to all invocations, but in second lambda, `init` will be instantiated for every call.
Idea is to avoid creating the device scalar multiple times since it is going to remain the same type.

Reply:
I was thinking that the `init` would be an argument of the lambda. The first case can create one `init` and the second case can create a different `init` for each type. The rest of the code is duplicated.

Reply:
updated. (calling lambda inside lambda).